### PR TITLE
allow resistance marker tables to show locus level granularity

### PR DESCRIPTION
**Pull request by bgpalmer**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206616195887915

### CODE REVIEW

**Comment by bgpalmer** [comment]
[resmarker_microhap_table.txt](https://github.com/EPPIcenter/mad4hatter/files/13183941/resmarker_microhap_table.txt)
[resmarker_new_mutations.txt](https://github.com/EPPIcenter/mad4hatter/files/13183942/resmarker_new_mutations.txt)
[resmarker_table.txt](https://github.com/EPPIcenter/mad4hatter/files/13183943/resmarker_table.txt)

**Comment by andres-ad** [comment]
Sorry it took me so long to look at this. This solves what we needed to solve!
Can you please modifiy the outputs so that markers show together (jsut cosmetic). e.g. k13  578 shows two times for each sample, they should be consecutive rows, right now all the alleles in one locus show together and then the following locus.

**Comment by bgpalmer** [comment]
@andres-ad does this look better?

[resmarker_microhap_table.txt](https://github.com/EPPIcenter/mad4hatter/files/14766002/resmarker_microhap_table.txt)
[resmarker_table.txt](https://github.com/EPPIcenter/mad4hatter/files/14766003/resmarker_table.txt)

**Comment by bgpalmer** [comment]
[resmarker_microhap_table.txt](https://github.com/EPPIcenter/mad4hatter/files/14782385/resmarker_microhap_table.txt)
[resmarker_table.txt](https://github.com/EPPIcenter/mad4hatter/files/14782386/resmarker_table.txt)

File: bin/resistance_marker_module.py [hunk 1]
```diff
@@ -256,24 +257,51 @@ def main(args):
...
     df_results = pd.DataFrame(results)
     df_results = df_results.rename(columns={'reads': 'Reads', 'sampleID': 'SampleID', "pseudo_cigar": "PseudoCIGAR"})
+    df_results['CodonID'] = df_results['CodonID'].astype(int)
 
-    df_results = df_results.sort_values(['CodonID', 'Gene', 'SampleID'] ,ascending=[False, False, False])
-    df_results = df_results[['SampleID', 'GeneID', 'Gene', 'CodonID', 'RefCodon', 'Codon', 'CodonStart', 'CodonRefAlt', 'RefAA', 'AA', 'AARefAlt', 'Reads', 'Locus', 'PseudoCIGAR', 'new_mutations']]
-    df_resmarker = df_results.drop(['Locus', 'PseudoCIGAR', 'new_mutations'], axis=1)
+    sort_columns, sort_order, select_columns = None, None, None
+    if args.add_locus_column:
+        sort_columns, sort_order = ['SampleID', 'chr', 'CodonID'], [True, True, True]
```
Comment (kathryn1995) [inline bin/resistance_marker_module.py:264]: you could take this line out of the if, else logic

**Comment by kathryn1995** [comment]
Outputs look great now, but it would be good to just always have the locus column present, not only if someone sets a flag as we would always want this output where we have tiling to include the locus

**Comment by kathryn1995** [comment]
Looks good!

**Pull request merged**